### PR TITLE
Adding delete_branch_on_merge support for repos

### LINKED ISSUE #1245: Support deleting head branch of GitHub pull requests
add support for this new feature: https://help.github.com/en/articles/managing-the-automatic-deletion-of-branches

### PR DESCRIPTION
**Pull request by stefansedich**
Adding support for the new delete_branch_on_merge setting for repositories.

"Finally" Closes #1245

### CODE REVIEW

File: github/repos.go [hunk 1]
```diff
@@ -303,12 +304,13 @@ type createRepoRequest struct {
...
-	AllowSquashMerge  *bool   `json:"allow_squash_merge,omitempty"`
-	AllowMergeCommit  *bool   `json:"allow_merge_commit,omitempty"`
-	AllowRebaseMerge  *bool   `json:"allow_rebase_merge,omitempty"`
+	AutoInit            *bool   `json:"auto_init,omitempty"`
+	GitignoreTemplate   *string `json:"gitignore_template,omitempty"`
+	LicenseTemplate     *string `json:"license_template,omitempty"`
+	AllowSquashMerge    *bool   `json:"allow_squash_merge,omitempty"`
+	AllowMergeCommit    *bool   `json:"allow_merge_commit,omitempty"`
+	AllowRebaseMerge    *bool   `json:"allow_rebase_merge,omitempty"`
+	DeleteBranchOnMerge *bool   `json:"delete_branch_on_merge,omitempty"`
```
Comment (gmlewis) [inline github/repos.go]: Please copy this new field's value after line 347 below, otherwise it has no effect.

Comment (stefansedich) [reply github/repos.go]: My bad :disappointed: my grep had failed me finding the places the other fields were set I assumed it was some magic but missed this obvious place! Should be fixed.

Comment (gmlewis) [reply github/repos.go]: No problem!  That's why we do code reviews!  :smile:
That's nothing compared to some of the things I've done.  :smile:
Thank you for fixing it!

**Comment by gmlewis** [comment]
Thank you, @wesleimp!
Merging.

**Pull request merged**